--- Controller.pm_orig_v3.5.1   2018-09-24 22:55:10.000000000 -0400
+++ Controller.pm_patched_20181109      2018-11-09 17:42:31.263404614 -0500
@@ -470,7 +470,19 @@

     if ($msgtype eq Amanda::Chunker::Protocol::DONE ||
        $msgtype eq Amanda::Chunker::Protocol::PARTIAL) {
-       $msg_params{'size'} = ($params{'data_size'}+0) / 1024;
+       # the message 'size' parameter is in integer kb, but for
+       # small-but-not-empty files round it up to 1 (since
+       # the driver process assumes a zero value indicates the dump was
+       # invalid).
+       my $data_size = ($params{'data_size'}+0);
+       if ($data_size > 0 and $data_size < 1024)  {
+           $msg_params{'size'} = 1;
+           # also update the "stats" string generated earlier in this
+           # function, to match the rounded-up size.
+           $stats = make_chunker_stats(1024, $params{'total_duration'});
+       } else {
+           $msg_params{'size'} = $data_size / 1024;
+       }
        $msg_params{'server_crc'} = $self->{'server_crc'};
        $msg_params{'stats'} = $stats;
     }

